src/fidasim.f90

```diff
@@ -2210,6 +2229,229 @@ subroutine make_beam_grid
+
+        !!Close SPEC group
+        call h5gclose_f(gid, error)
+    endif
+
+    !!If NPA group exists, read NPA LOS info 
+    call h5ltpath_valid_f(fid, "/npa", .True., path_valid_npa, error)
+    if(.not.path_valid_npa) then
+        if(inputs%verbose.ge.0) then
+            write(*,'(a)') 'NPA geometry is not in the geometry file'
+            write(*,'(a)') 'Passive NPA calculations will not be performed'
+        endif
+        inputs%calc_pnpa = 0
+    endif
+
+    if(inputs%calc_pnpa.gt.0) then
+        !!Open NPA group
+        call h5gopen_f(fid, "/npa", gid, error)
+
+        call h5ltread_dataset_int_scalar_f(gid, "/npa/nchan", npa_nchan, error)
+
+        allocate(a_cent(3,  npa_nchan), d_cent(3,  npa_nchan))
+
+        !! Read in a_cent and d_cent info
+        dims = [3,spec_nchan]
+        call h5ltread_dataset_double_f(gid, "/npa/a_cent",  a_cent, dims, error)
+        call h5ltread_dataset_double_f(gid, "/npa/d_cent",  d_cent, dims, error)
+
+        !!Close NPA group
+        call h5gclose_f(gid, error)
+    endif
+
+    !!Close file id
+    call h5fclose_f(fid, error)
+
+    !!Close HDF5 interface
+    call h5close_f(error)
+
+    !!Collect all LOS info
+    allocate(r0_arr(3,  spec_nchan+npa_nchan), v0_arr(3,  spec_nchan+npa_nchan))
+
+    if((inputs%calc_pfida.gt.0).and.(inputs%calc_pnpa.gt.0)) then
+        r0_arr(:,1:spec_nchan) = lenses
+        v0_arr(:,1:spec_nchan) = axes
+        r0_arr(:,(1+spec_nchan):(spec_nchan+npa_nchan)) = d_cent
+        v0_arr(:,(1+spec_nchan):(spec_nchan+npa_nchan)) = a_cent - d_cent
+    else if(inputs%calc_pfida.gt.0) then
+        r0_arr(:,1:spec_nchan) = lenses
+        v0_arr(:,1:spec_nchan) = axes
+    else !pnpa>=1 case
+        r0_arr(:,1:npa_nchan) = d_cent
+        v0_arr(:,1:npa_nchan) = a_cent - d_cent
+    endif
+
+    !! The remainder of the code will make the passive neutral grid
+    pass_grid%dr = inter_grid%dr
+    pass_grid%dz = inter_grid%dz
+    pass_grid%nr = inter_grid%nr
+    pass_grid%nz = inter_grid%nz
+    allocate(pass_grid%r(pass_grid%nr), pass_grid%z(pass_grid%nz))
+    pass_grid%r = inter_grid%r
+    pass_grid%z = inter_grid%z
+    pass_grid%da = pass_grid%dr*pass_grid%dz
+
+    !! Start by finding the limits of the beam grid vertices to define the phi information
+    xmin = beam_grid%xmin ; xmax = beam_grid%xmax
+    ymin = beam_grid%ymin ; ymax = beam_grid%ymax
+    zmin = beam_grid%zmin ; zmax = beam_grid%zmax
+
+    vertex111(1) = xmin   ; vertex111(2) = ymin   ; vertex111(3) = zmin
+    vertex222(1) = xmax   ; vertex222(2) = ymax   ; vertex222(3) = zmax
+
+    vertex112 = vertex111 ; vertex211 = vertex111 ; vertex121 = vertex111
+    vertex112(3) = zmax   ; vertex211(1) = xmax   ; vertex121(2) = ymax
+
+    vertex122 = vertex222 ; vertex221 = vertex222 ; vertex212 = vertex222
+    vertex122(1) = xmin   ; vertex221(3) = zmin   ; vertex212(2) = ymin
+
+    call xyz_to_uvw(vertex111,vertex111_uvw)
```

Comment:
Why isn't there a `xyz_to_cyl` and a `cyl_to_xyz` subroutine? You could reduce the size of this section by a factor of two